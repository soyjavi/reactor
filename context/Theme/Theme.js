import { node, shape } from 'prop-types';
import React, {
  createContext, useContext, useEffect, useState,
} from 'react';

import THEME from '../../common/theme';

const ThemeContext = createContext('reactor:theme');

const ThemeProvider = ({ children, style }) => {
  const [theme, setTheme] = useState(THEME.extend(style));
  useEffect(() => {
    setTheme(THEME.extend(style));
  }, [style]);

  return (
    <ThemeContext.Provider value={theme}>
      { children }
    </ThemeContext.Provider>
  );
};

ThemeProvider.propTypes = {
  children: node,
  style: shape({}),
};

ThemeProvider.defaultProps = {
  children: undefined,
  style: {},
};

export { ThemeProvider };

export const useTheme = () => useContext(ThemeContext);
