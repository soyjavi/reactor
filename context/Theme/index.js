export { ThemeProvider, useTheme } from './Theme';
