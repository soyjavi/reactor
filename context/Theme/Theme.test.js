import React from 'react';
import TestRenderer from 'react-test-renderer';

import { ThemeProvider } from './Theme';
import THEME from '../../common/theme';

const customTheme = { COLOR: { PRIMARY: 'green' } };

describe('<ThemeProvider>', () => {
  it('renders', () => {
    const tree = TestRenderer.create(<ThemeProvider />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    expect(THEME.COLOR.PRIMARY).toEqual('#10c0bc');
    const tree = TestRenderer.create(<ThemeProvider style={customTheme} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(THEME.COLOR.PRIMARY).toEqual(customTheme.COLOR.PRIMARY);
  });
});
