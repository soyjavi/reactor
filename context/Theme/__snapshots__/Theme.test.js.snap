// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeProvider> renders 1`] = `null`;

exports[`<ThemeProvider> when {style} 1`] = `null`;
